_posts/2014-03-23-disjoint-sets-and-agglomerative-clustering.md

```diff
@@ -31,50 +31,50 @@ Disjoint-sets provide a ready-made solution to this problem. The data structure
 
 Three basic operations are supported by disjoint-sets:
 
-- **Make-Set(x)** creates a new set containing only x
-- **Union(x, y)** combines x and y into a single set
-- **Find(x)** identifies the set to which x belongs
+- **make_set(x)** creates a new set containing only x
+- **union(x, y)** combines x and y into a single set
+- **find(x)** identifies the set to which x belongs
 
-Because of the importance of Union(x, y) and Find(x), disjoint-sets are sometimes referred to as union-find data structures.
+Because of the importance of union(x, y) and find(x), disjoint-sets are sometimes referred to as union-find data structures.
 
 Disjoint-sets, like the initial hash table approach, can be implemented as a mapping of element → subset. A particular subset is uniquely identified by the root element in the tree representing that subset. And an element is a root element if it is its own parent.
 
 ![disjoint set representation](/assets/images/2014-03-23/disjoint_set_representation.png)
 An example of {A, B, C, D} at after the second merge. Only {A, B, C} and {D} remain.
 {: .caption}
 
-In a neat bit of symmetry with bottom-up clustering, disjoint-sets are initialised by calling Make-Set on each element.
+In a neat bit of symmetry with bottom-up clustering, disjoint-sets are initialised by calling make_set on each element.
 
 {% highlight python %}
-def MakeSets(x):
+def make_sets(x):
     return dict([(i, i) for i in x])
 {% endhighlight %}
 
 To find the set to which a particular element belongs, we recursively look for the parent until we find a node that is its own parent.
 
 {% highlight python %}
-def Find(x, parent):
-    if parent[x] is x:
```

Comment:
This is an important distinction. `is` really should only be used to compare None, True, and False, as it asserts that the two things being compared point to the same memory address. Whereas `==` asserts that the two things being compared have the same value. Here's a great example:
```
>>> a is 1
True
>>> b = 500
>>> b is 500
False
```
Python caches small numbers in memory for convenience sake (0-255), and `is` works for them. But once you get to 256, bad things happen.